consumer/services.go

```diff
@@ -555,3 +561,53 @@ func (p *paymentTokens) pubSub(channel string, message interface{}) {
 		fmt.Println(msg.Channel, msg.Payload)
 	}
 }
+
+//send sms to the user
+func (s *Service) SendSMS(c *gin.Context) {
+	// from gin, you read the request body (api documentations TODO)
+	var fields SMS
+	//var fields ebs_fields.MobileRedis
+	err := c.ShouldBindWith(&fields, binding.JSON)
+
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error(), "code": "unmarshalling_error"})
+	}
+	var message string
+	phone := c.GetString("phone")
+	message = c.GetString("message")
+	if (phone == "") && (message == "") {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error(), "code": "invalid values"})
+	}
+	fields.MobileNo = phone
+	fields.message = message
```

Comment:
Nope, this doesn't work like this. You have to specify how your api endpoint would be used. `c.ShouldBindWith` binds the request body to a json object -- and only works for `POST` requests.

`c.GetString` get's a string from the context -- works when we want to by-pass data between middlewares, check `c.GetString("token")`.

Rewrite this code to something like this:

if err := c.ShouldBindWith // previous comment
// handle errors properly
if err := sendSms(fields) // we marshalled it earlier

Reply:
this whole part is not needed